homeassistant/components/sensor/wunderground.py

```diff
@@ -663,7 +668,7 @@ def __init__(self, hass: HomeAssistantType, rest, condition):
         self._entity_picture = None
         self._unit_of_measurement = self._cfg_expand("unit_of_measurement")
         self.rest.request_feature(SENSOR_TYPES[condition].feature)
-        self.entity_id = generate_entity_id(
+        self.entity_id = async_generate_entity_id(
```

Comment:
All other platforms using `async_generate_entity_i`d this way, but should we really be using an event loop callback from within `__init__`? Would `async_added_to_hass` also work or is the entity_id required before that method is called?

Reply:
Why not? It's not doing any I/O, so should be ok.

Reply:
In order to call an `async_*` function in Home Assistant we need to guarantee that the caller is inside the event loop. If we can't guarantee that, we usually wrap the async call inside a `hass.add_job()`.

So, because of our direct call to `async_generate_entity_id()` we'd also need to guarantee that `__init__` is executed from within the event loop. Can we guarantee that though? In this case, it's only used in `asyn_setup_platform` (therefore inside the event loop), but I mean the object *could* also be initialized from outside the event loop.

Reply:
Entities are supposed to be initialized in setup_platform/async_setup_platform. Also the caller is responsible to properly handle the objects it calls. So it's fine like this.

Reply:
`async_added_to_hass` is executed after entity_id is assigned. This is fine since the platform uses `async_setup_platform`